### PR TITLE
English translation is updated

### PR DESCRIPTION
**Pull request by BayoNet**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

**Comment by alexey-milovidov** [comment]
Thanks!
BTW, the files became executable. This is bad but tolerable.

File: docs/en/dicts/external_dicts_dict.md [hunk 1]
```diff
@@ -27,7 +27,8 @@ The dictionary configuration has the following structure:
 ```
 
 - name – The identifier that can be used to access the dictionary. Use the characters `[a-zA-Z0-9_\-]`.
-- [source](external_dicts_dict_sources.md#dicts-external_dicts_dict_sources) – Source of the dictionary.
-- [layout](external_dicts_dict_layout.md#dicts-external_dicts_dict_layout) – Location of the dictionary in memory.
-- [structure](external_dicts_dict_structure.md#dicts-external_dicts_dict_structure) – Structure of the dictionary. A key and attributes that can be retrieved by this key.
-- [lifetime](external_dicts_dict_lifetime.md#dicts-external_dicts_dict_lifetime) – How frequently to update dictionaries.
+- [source](external_dicts_dict_sources.html/#dicts-external_dicts_dict_sources) — Source of the dictionary .
```
Comment (filimonov) [inline docs/en/dicts/external_dicts_dict.md]: Wrong url. Inaccurate master merge.

File: docs/en/dicts/external_dicts_dict.md [hunk 2]
```diff
@@ -27,7 +27,8 @@ The dictionary configuration has the following structure:
...
+- [layout](external_dicts_dict_layout.md#dicts-external_dicts_dict_layout) — Dictionary layout in memory.
+- [source](external_dicts_dict_sources.html/#dicts-external_dicts_dict_sources) — Structure of the dictionary . A key and attributes that can be retrieved by this key.
```
Comment (filimonov) [inline docs/en/dicts/external_dicts_dict.md]: Same

File: docs/en/dicts/external_dicts_dict_layout.md [hunk 3]
```diff
@@ -227,66 +226,4 @@ Do not use ClickHouse as a source, because it is slow to process queries with ra
 
 ### complex_key_cache
 
-This type of storage is designed for use with compound [keys](external_dicts_dict_structure#dicts-external_dicts_dict_structure). Similar to `cache`.
-
-<a name="dicts-external_dicts_dict_layout-ip_trie"></a>
-
-### ip_trie
```
Comment (filimonov) [inline docs/en/dicts/external_dicts_dict_layout.md]: Ip_tree Fragment removed

File: docs/en/dicts/external_dicts_dict_structure.md [hunk 4]
```diff
@@ -63,12 +63,10 @@ Configuration fields:
...
 
-The key can be a `tuple` from any types of fields. The [ layout](external_dicts_dict_layout.md#dicts-external_dicts_dict_layout)  in this case must be `complex_key_hashed`  or `complex_key_cache`.
+The key can be a `tuple` from any types of fields. The [layout](external_dicts_dict_layout.md#dicts-external_dicts_dict_layout) in this case must be `complex_key_hashed` or `complex_key_cache`.
 
 <div class="admonition tip">
-
-A composite key can also consist of a single element, which makes it possible to use a string as the key, for instance.
-
+A composite key can consist of a single element.
 This makes it possible to use a string as the key, for instance.
```
Comment (filimonov) [inline docs/en/dicts/external_dicts_dict_structure.md]: Esc-M?

File: docs/en/dicts/external_dicts_dict_structure.md [hunk 5]
```diff
@@ -119,6 +117,6 @@ Configuration fields:
 - `null_value` – The default value for a non-existing element. In the example, it is an empty string.
 - `expression` – The attribute can be an expression. The tag is not required.
 - `hierarchical` – Hierarchical support. Mirrored to the parent identifier. By default, ` false`.
-- `injective` Whether the `id -> attribute` image is injective. If ` true`, then you can optimize the ` GROUP BY` clause. By default, `false`.
-- `is_object_id` - Used for query mongo documents by ObjectId
+- `injective` – Whether the `id -> attribute` image is injective. If ` true`, then you can optimize the ` GROUP BY` clause. By default, `false`.
+- `is_object_id` – Whether the query is executed for a MongoDB document by `ObjectID`.
```
Comment (filimonov) [inline docs/en/dicts/external_dicts_dict_structure.md]: Not accurate

File: docs/en/functions/array_functions.md [hunk 6]
```diff
@@ -39,7 +39,7 @@ Accepts an empty array and returns a one-element array that is equal to the defa
 Returns an array of numbers from 0 to N-1.
 Just in case, an exception is thrown if arrays with a total length of more than 100,000,000 elements are created in a data block.
 
-## array(x1, ...), operator \[x1, ...\]
+## array(x1, ...), оператор \[x1, ...\]
```
Comment (filimonov) [inline docs/en/functions/array_functions.md]: Ups

File: docs/en/functions/json_functions.md [hunk 7]
```diff
@@ -1,11 +1,11 @@
-# Functions for working with JSON.
+# Functions for working with JSON
 
 In Yandex.Metrica, JSON is transmitted by users as session parameters. There are some special functions for working with this JSON. (Although in most of the cases, the JSONs are additionally pre-processed, and the resulting values are put in separate columns in their processed format.) All these functions are based on strong assumptions about what the JSON can be, but they try to do as little as possible to get the job done.
 
 The following assumptions are made:
 
 1. The field name (function argument) must be a constant.
-2. The field name is somehow canonically encoded in JSON. For example: `visitParamHas('{"abc":"def"}', 'abc') = 1`, but `visitParamHas('{"\\u0061\\u0062\\u0063":"def"}', 'abc') = 0`
+2. The field name is somehow canonically encoded in JSON. For example: `visitParamHas('{"abc":"def"}', 'abc') = 1`, но `visitParamHas('{"\\u0061\\u0062\\u0063":"def"}', 'abc') = 0`
```
Comment (filimonov) [inline docs/en/functions/json_functions.md]: ?

File: docs/en/getting_started/index.md [hunk 8]
```diff
@@ -16,14 +16,15 @@ The terminal must use UTF-8 encoding (the default in Ubuntu).
...
 For testing and development, the system can be installed on a single server or on a desktop computer.
 
-### Installing from packages Debian/Ubuntu
+### Installing from packages
 
 In `/etc/apt/sources.list` (or in a separate `/etc/apt/sources.list.d/clickhouse.list` file), add the repository:
 
 ```text
-deb http://repo.yandex.ru/clickhouse/deb/stable/ main/
+deb http://repo.yandex.ru/clickhouse/trusty stable main
```
Comment (filimonov) [inline docs/en/getting_started/index.md]: Ups

File: docs/en/interfaces/http_interface.md [hunk 9]
```diff
@@ -194,11 +191,7 @@ $ echo 'SELECT number FROM system.numbers LIMIT 10' | curl 'http://localhost:812
...
-
-You can get the progress of query execution in X-ClickHouse-Progress headers, by enabling setting send_progress_in_http_headers.
-
-Running query are not aborted automatically after closing HTTP connection. Parsing and data formatting are performed on the server side, and using the network might be ineffective.
+In contrast to the native interface, the HTTP interface does not support the concept of sessions or session settings, does not allow aborting a query (to be exact, it allows this in only a few cases),  and does not show the progress of query processing. Parsing and data formatting are performed on the server side, and using the network might be ineffective.
```
Comment (filimonov) [inline docs/en/interfaces/http_interface.md]: Ups

**Comment by filimonov** [comment]
Very innacurate master merge before push. Lot of regressions

**Comment by filimonov** [comment]
P.s. I didn't review all the changes. Probably there are more problems.

**Comment by silviucpp** [comment]
Hello guys,

Are you sure about : "In contrast to the native interface, the HTTP interface does not support the concept of sessions or session settings, does not allow aborting a query (to be exact, it allows this in only a few cases)". I mean was this functionality removed ? because I'm using session_id, session_timeout and some other settings over http and works nice (I'm using a branch a bit older from december).

**Comment by filimonov** [comment]
@silviucpp it looks like regression of #2037 (and few other doc commits) caused by inaccurate conflict resolution during merge.

**Comment by silviucpp** [comment]
@filimonov great :) I thought the features were removed :)

**Pull request merged**